api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -25,7 +25,23 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.transport_sockets.tls]
 // The TLS contexts below provide the transport socket configuration for upstream/downstream TLS.
 
-// [#next-free-field: 8]
+// [#next-free-field: 3]
+// TLS per host session cache configuration.
+message TlsPerHostSessionCacheConfig {
+  // Specifies the maximum number of upstream hosts that can have their TLS session keys cached.
+  // This is useful when different upstream hosts serving the same destination use distinct TLS
+  // configurations, meaning a session key from one host isn't reusable with another.
+  //
+  // If not explicitly set, the default is to cache session keys for 10 upstream hosts.
+  google.protobuf.UInt32Value max_hosts = 1;
```

Comment:
Is 0 a valid value for either of these fields?  I assume not, in which case they can probably use `uint32` instead of `google.protobuf.UInt32Value`, because we can treat 0 as unset.

Reply:
Makes sense, updated pr to use `uint32`.